components/widgets/ui/src/main/resources/PhenoTips/GeneVariantMacros.xml

```diff
@@ -0,0 +1,1420 @@
+        <prettyName>Use this extension</prettyName>
+        <relationalStorage>0</relationalStorage>
+        <separator> </separator>
+        <separators>|, </separators>
+        <size>1</size>
+        <unmodifiable>0</unmodifiable>
+        <values>currentPage|onDemand|always</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </use>
+    </class>
+    <property>
+      <cache>long</cache>
+    </property>
+    <property>
+      <code>var ExtraGeneVariantData = (function (ExtraGeneVariantData) {
+  var tools = ExtraGeneVariantData.tools = ExtraGeneVariantData.tools || {};
+  tools.Editor = Class.create({
+    initialize : function () {
+      this.geneClassName = $$('.gene-table.extradata-list')[0].id.substring($$('.gene-table.extradata-list')[0].id.lastIndexOf('-')+1);
+      this.geneVariantClassName = 'PhenoTips.GeneVariantClass';
+      //if edit mode
+      if (!!$('inline')) {
+        this.warnSaving = false;
+        this.getNewRowsTemplates();
+        this.createEditButtons('.variant-moreinfo-editbutton-row');
+        this.createEditDoneButtons('.variant-moreinfo-editdonebutton-row');
+        $$('.gene-table a.variant-edit').invoke('observe', 'click', this.editData.bindAsEventListener(this));
+        $$('.gene-table a.variant-edit-done').invoke('observe', 'click', this.editDoneData.bindAsEventListener(this));
+        $$('.variant.moreinfo').invoke('observe', 'click', this.areaEditData.bindAsEventListener(this));
+        $$('.gene a.delete-gene').invoke('observe', 'click', this.ajaxDeleteGeneData.bindAsEventListener(this));
+        $$('a.button.add-gene.add-data-button').invoke('observe', 'click', this.ajaxAddGeneData.bindAsEventListener(this));
+        $$('.variant a.delete-variant').invoke('observe', 'click', this.ajaxDeleteVariantData.bindAsEventListener(this));
+        $$('a.button.add-variant.add-data-button').invoke('observe', 'click', this.ajaxAddVariantData.bindAsEventListener(this));
+        //invoking click event to hide rows with empty inputs in 'more info' section
+        $$('.gene-table a.variant-edit-done').invoke('click');
+        this.lockGeneInput();
+
+      }
+      $$('.gene-table tr tr').each( function(item) {
+        item.toggleClassName('moreinfo-view', true);
+      });
+      this.createShowMoreinfoButtons('td.variant-row-count.variant');
+      this.createHideShowButtons('.variant.variant-title');
+      $$('.gene-table tr th')[0].width = '1em';
+    },
+    areaEditData  : function (event) {
+      var el = null;
+      if (event.element().className == 'variant moreinfo')
+        el = event.element();
+      else
+        el = event.element().up('td.variant.moreinfo');
+      el.select('a.variant-edit').invoke('click');
+      var className = el.down('.variant-moreinfo-editbutton-row').className;
+      var varIndex = className.substring(className.lastIndexOf('-')+1);
+      event.stopPropagation();
+      document.observe('click', function (event) {
```

Comment:
This listener hangs around forever, so if you call `areaEditData` multiple times, there would be multiple listeners attached. Not sure if this was intended.